### ISSUE TITLE
BandInput tests appear to be broken

### ISSUE THREAD

**Issue by wlycdgrfromflatiron** [learn-co-curriculum/building-forms-lab#14]
As written, the BandInput tests that use `const wrapper = shallow(<BandInput />)` throw an "Invariant Violation: could not find 'store' in either the context or the props...." error. When corrected by passing in the store as a prop, they still don't work -- apparently because you gotta .dive() to get down past the Redux connect() wrapper and into the component ([HOCs and .dive()](https://github.com/airbnb/enzyme/issues/1002))

This works:
```
    const store = createStore(manageBand)
    const wrapper = shallow(<BandInput store={store} />).dive()
```

**Comment by LaneMiller** [comment]
The alternative here is using the undecorated component via a named export, as is recommended in the [docs](https://redux.js.org/recipes/writing-tests#connected-components):

BandInput.js
```
export class BandInput extends Component { ... }

export default connect(null, mapDispatchToProps)(BandInput)
```

index-tests.js
```
import { BandInput } from '../src/components/BandInput';
```

**Comment by maxwellbenton** [comment]
Hi @wlycdgrfromflatiron,

Thank you for providing this feedback. I've tested the changes you made but unfortunately, they don't work with the existing solution. I've also taken a look at your solution - as per the readme, `BandInput` should be a child component of `BandsContainer`. `BandsContainer` is meant to be the only component that connects Redux to both the state and dispatch, passing the appropriate props to `BandInput` and `Bands`. In your solution, `BandInput` connects to dispatch on its own, as a sibling of `BandsContainer`. The tests treat `BandInput` as a regular, unconnected React component that just handles props and doesn't need to know about `store`.

I've opened a separate issue documented a needed change to this lesson's tests to ensure the correct component structure is being used prior to testing for access to the store. I'm going to go ahead and close this issue.

Thank you for contributing to a learn.co lesson!

**Comment by maxwellbenton** [comment]
Hi @wlycdgr,

Following up on this issue. We've updated this lab to include more guidance on how to structure the application, specifically regarding where Redux should be connected. As the lab currently is, the tests can't really be more flexible on where Redux is connected. Instead, we've added an additional test that makes it clearer that the `BandsContainer` is where the connection to Redux is made, and that the action to add a band should be made available to `BandInput` via props, not through a direct connection to dispatch. This should help avoid confusion for students in the future and prevent them from building out a valid application only to find it doesn't align with the tests.

Thank you for providing this feedback and helping to improve our curriculum!
